sdb/commands/zfs/dbuf.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class DslDir():
+
+    @staticmethod
+    def name(dd):
+        pname = ''
+        if dd.dd_parent:
+            pname = DslDir.name(dd.dd_parent) + '/'
+        return pname + dd.dd_myname.string_().decode("utf-8")
+
+
+class Dataset():
+
+    @staticmethod
+    def name(ds):
+        if not ds:
+            return 'MOS'
+        name = DslDir.name(ds.ds_dir)
+        if not ds.ds_prev:
+            sn = ds.ds_snapname.string_().decode("utf-8")
+            if len(sn) == 0:
+                sn = '%UNKNOWN_SNAP_NAME%'
+            name += '@' + sn
+        return name
+
+
+class Objset():
+
+    @staticmethod
+    def name(os):
+        return Dataset.name(os.os_dsl_dataset)
```

Comment:
For your question about the organization of these: What I'd personally do is have them here as you do initially, then when more consumers show up put these classes under `sdb/commands/zfs/internal/zfs_structures.py` (or whatever name). The reasoning for having them in a separate file is just so we don't introduce unintuitive dependencies between commands. The reasoning of having them all in a single file is even more subjective - I personally prefer having one file with multiple related classes that takes 100 LoC vs having say 4 files, each with 20 LoC.

Again, the whole codebase is pretty small so it really doesn't matter. I just wrote the above since you asked.

Reply:
After fighting with pylint, I moved these to helper funcs in the `Dbuf` class.  We should probably revisit this once we need more helpers for these types.